### PR TITLE
docker: Refactor of buildDocker.sh

### PR DESCRIPTION
**Pull request by Willsparker**
Ref: 
https://ci.adoptopenjdk.net/view/Tooling/job/DockerfileCheck/159/
https://ci.adoptopenjdk.net/view/Tooling/job/DockerfileCheck/162/
(The job for both of these runs had the config changed to run from this branch to test it)

A big refactor of `buildDocker.sh` changing the following:
- Rename variables, replace duplicated file paths with variables, add whitespace where it makes the code more readable
- Use the same logic in `dockerfile-generator.sh` ( using the API ) to set `JDK_MAX` and `JDK_GA`, so the script doesn't have to be changed in the future to add in later JDKs.
- The script runs within the openjdk-build directory it's executed, instead of re-cloning the openjdk-build repository again (requested by @sxa )
- Removed the `all` option as it seemed superfluous.

### CODE REVIEW

File: docker/buildDocker.sh [hunk 1]
```diff
@@ -43,135 +64,102 @@ parseCommandLineArgs()
...
-		- jdk9u
-		- jdk10u
-		- jdk11u
-		- jdk12u
-		- jdk13u
-		- jdk14u
-		- jdk"
+checkJDK() {
+  if ! ((JDK_VERSION >= 8 && JDK_VERSION <= JDK_MAX)); then
+    echo "Please input a JDK between 8 & ${JDK_MAX}, or 'jdk'"
```
Comment (karianna) [inline docker/buildDocker.sh:77]: a concrete example will help here

File: docker/buildDocker.sh [hunk 2]
```diff
@@ -1,13 +1,30 @@
 #!/bin/bash
 set -u
 
-jdkVersion=''
-bootDir=''
-buildVariant="hotspot"
+BOOTDIR=''
```
Comment (karianna) [inline docker/buildDocker.sh:4]: single quotes used here and double quotes on the next line - should we be consistent here?

File: docker/buildDocker.sh [hunk 3]
```diff
@@ -19,18 +36,22 @@ parseCommandLineArgs()
...
+					shift;;
 				"--jdk-boot-dir" | "-J")
-					bootDir="$1"; shift;;
+					BOOTDIR="$1"; shift;;
 				"--openj9" | "-j9")
-					buildVariant="openj9";;
+					VARIANT="openj9";;
 				"--use-eclipse-docker-files" | "-e" )
-					useEclipseOpenJ9DockerFiles=true; buildVariant="eclipse";;
+					useEclipseOpenJ9DockerFiles=true; VARIANT="eclipse";;
```
Comment (karianna) [inline docker/buildDocker.sh:54]: is variant actually openj9?

File: docker/buildDocker.sh [hunk 4]
```diff
@@ -43,135 +64,102 @@ parseCommandLineArgs()
...
+	local dockerfileDir="$WORKSPACE/EclipseDockerfiles"
+	local jdk="jdk"
+
+	mkdir -p $dockerfileDir
+	cd $dockerfileDir
 	wget https://raw.githubusercontent.com/eclipse/openj9/master/buildenv/docker/mkdocker.sh
 	chmod +x mkdocker.sh
 	# Generate an Ubuntu1804 Dockerfile using mkdocker.sh
-	$WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles/mkdocker.sh --dist=ubuntu --version=18 --print >> $WORKSPACE/DockerBuildFolder/$jdkVersion-$buildVariant/EclipseDockerfiles/Dockerfile
+	$dockerfileDir/mkdocker.sh --dist=ubuntu --version=18 --print >> $dockerfileDir/Dockerfile
```
Comment (karianna) [inline docker/buildDocker.sh:110]: do we want to pass in 18 as a parameter?

Comment (Willsparker) [reply docker/buildDocker.sh:110]: https://github.com/AdoptOpenJDK/openjdk-build/pull/1911#discussion_r444010951 
For now, it's hardcoded to stay `18` as all of the Dockerfiles generated using `dockerfile-generator.sh` can only be Ubuntu1804, so I think the `mkDocker.sh` generated dockerfiles should be consistent with those :-)

File: docker/buildDocker.sh (hunk 3 above)
Comment (Willsparker) [reply docker/buildDocker.sh]: Fixed with latest commit

File: docker/buildDocker.sh (hunk 2 above)
Comment (Willsparker) [reply docker/buildDocker.sh]: Fixed with latest commit

File: docker/buildDocker.sh (hunk 1 above)
Comment (Willsparker) [reply docker/buildDocker.sh:77]: Fixed with latest commit

**Comment by Willsparker** [comment]
@karianna Yep, it's been through `dockerfileCheck` a few times to check it's good

File: docker/buildDocker.sh [hunk 5]
```diff
@@ -1,13 +1,30 @@
...
+WORKSPACE=${PWD}
+JDK_VERSION=
+JDK_MAX=
+JDK_GA=
 
-#takes in all arguments to determine script options
+# shellcheck disable=SC2002 # Disable UUOC error
+setJDKVars() {
+	wget -q https://api.adoptopenjdk.net/v3/info/available_releases
+	JDK_MAX=$(cat available_releases \
```
Comment (sxa) [inline docker/buildDocker.sh:16]: You could simplify this with awk which would reduce the number of processes by 3:
JDK_MAX=$(awk -F: '/tip_version/{gsub("[, ]","",$2); print$2}' < available_releases)

File: docker/buildDocker.sh [hunk 6]
```diff
@@ -43,135 +64,103 @@ parseCommandLineArgs()
...
-	ssh-keyscan github.com >> $PWD/known_hosts
+	ssh-keygen -q -f $dockerfileDir/id_rsa -t rsa -N ''
+	cat id_rsa.pub >> $dockerfileDir/authorized_keys
+	ssh-keyscan github.com >> $dockerfileDir/known_hosts
 
-	# ${jdk%?} will remove the 'u' from 'jdkXXu'.
-	for jdk in $jdkVersion
-	do
-		if [ ${jdk} != "jdk" ]; then
-			jdk=${jdk%?}
```
Comment (sxa) [inline docker/buildDocker.sh:126]: Any reason not to use `${jdk%u}` instead of just stripping anything?

File: docker/buildDocker.sh (hunk 5 above)
Comment (sxa) [inline docker/buildDocker.sh:16]: You could simplify this with awk which would reduce the number of processes by 3:
JDK_MAX=$(awk -F: '/tip_version/{gsub("[, ]","",$2); print$2}' < available_releases)

File: docker/buildDocker.sh (hunk 6 above)
Comment (Willsparker) [reply docker/buildDocker.sh:126]: That's a bit that's been deleted

File: docker/buildDocker.sh (hunk 5 above)
Comment (Willsparker) [reply docker/buildDocker.sh]: Tested it and committed it :-)

File: docker/buildDocker.sh (hunk 4 above)
Comment (sxa) [reply docker/buildDocker.sh:105]: @karianna Are you ok with that explanation? Would be good to merge this and test the merge overnight if possible

Comment (karianna) [reply docker/buildDocker.sh:105]: I'll live with it :-)

**Comment by sxa** [comment]
Testing at https://ci.adoptopenjdk.net/view/Tooling/job/DockerfileCheck/164/

**Comment by sxa** [comment]
@Willsparker The above run seems to have broken on JDK11u/J9 with the cmake libdwarf error :-(

**Comment by Willsparker** [comment]
@sxa `DFC` is still running on this branch - which I guess I made this PR before #1971 . Changing the config to test the `adoptopenjdk/master` branch :-) https://ci.adoptopenjdk.net/view/Tooling/job/DockerfileCheck/165/

**Pull request merged**